server.go

```diff
@@ -117,16 +118,17 @@ func (s *BaseServer) StartStreams(ctx context.Context, cb EventCallback) {
 // Start starts the workers. Workers will call cb with results of requests. the call will
 // feed requests to workers by feeding requests to channel.
 // panics if workers number is zero.
-func (s *BaseServer) Start(ctx context.Context, workers uint, cb Callback) chan<- *Request {
+func (s *BaseServer) Start(ctx context.Context, workers uint, cb Callback) (chan<- *Request, *sync.WaitGroup) {
```

Comment:
I kinda like to let the control over the WaitGroup to the caller. So he can maybe use it for other things.
So I wouldn't allocate the WaitGroup in this function but instead just accept it as an argument.

Reply:
hmm, sure